frontend/admin/src/js/components/user/UserTableFilterDialog.tsx

```diff
@@ -250,6 +263,42 @@ const UserTableFilterButton = ({
   const handleSubmit: SubmitHandler<FormValues> = (data) => {
     onSubmit(data);
     setActiveFilters(data);
+    // this state lives in the UserTable component, this step also acts like a formValuesToSubmitData function
+    onFilterChange({
+      applicantFilter: {
+        expectedClassifications: data.classifications.map((classification) => {
+          const splitString = classification.toString().split("-");
```

Comment:
I think we could prob adjust this type to only be string, which would let you clean this up a bit. We have some generic type variables of `Object<V>` but it's always string
```
# shows only string
git grep 'Option<[a-z]\+>'
```

For no particular reason I dragged forward an Option type from earlier that allowed key to be either string or number:
https://github.com/GCTC-NTGC/gc-digital-talent/blob/71ded63044636b25c0b5b0ccaad7a106eec85e34/frontend/common/src/components/form/Select/SelectFieldV2.tsx#L11

But I don't think we ever use a number, except in the stories for SelectField:

https://github.com/GCTC-NTGC/gc-digital-talent/blob/71ded63044636b25c0b5b0ccaad7a106eec85e34/frontend/common/src/components/form/Select/SelectFieldV2.stories.tsx#L38-L40

https://github.com/GCTC-NTGC/gc-digital-talent/blob/71ded63044636b25c0b5b0ccaad7a106eec85e34/frontend/common/src/components/form/MultiSelect/MultiSelectFieldV2.stories.tsx#L38-L40

Reply:
yeah, I didn't want to be too destructive when it came to adding this pull request in so I tried to just adjust things to fit
typing changes https://github.com/GCTC-NTGC/gc-digital-talent/pull/3893/commits/70f4cfb7afae25937fa4320f5618497948611318